src/shogun/lib/DynamicObjectArray.h

```diff
@@ -413,13 +301,16 @@ class CDynamicObjectArray : public CSGObject
 		}
 
 		/** @return underlying array of pointers */
-		inline CSGObject** get_array() const { return m_array.get_array(); }
+		SG_FORCED_INLINE auto get_array() { return m_array.data(); }
 
 		/** shuffles the array (not thread safe!) */
-		inline void shuffle() { m_array.shuffle(); }
+		SG_FORCED_INLINE void shuffle() { std::random_shuffle(m_array.begin(), m_array.end()); }
 
 		/** shuffles the array with external random state */
-		inline void shuffle(CRandom * rand) { m_array.shuffle(rand); }
+		SG_FORCED_INLINE void shuffle(CRandom * rand)
+		{
+		//	m_array.shuffle(rand);
```

Comment:
Is that intentional?

Reply:
yes yes... i'm more wondering just about the implementation of clone